### ISSUE TITLE
ADMM Hyper-parameters

### ISSUE THREAD

**Issue by utsav-akhaury** [utsav-akhaury/ADMM#3]
Current ADMM hyper-parameters lead to an optimal reconstruction with NMSE ~7%, the set of choices being taken from [this implementation](https://github.com/CosmoStat/ShapeDeconv/blob/a18bf74fa7104f8621871c6ced61c503a4ec004a/python/DeepDeconv/admm/admm_FCS.py#L31) of the [ADMM Deep Plug&Play paper](https://arxiv.org/pdf/1911.00443.pdf).

Ideally, after reaching the optimal reconstruction, the algorithm should converge and give the same output for subsequent iterations. However, in both the NumPy and TensorFlow implementations, the solution diverges after a certain number of iterations (as shown in this [notebook](https://github.com/utsav-akhaury/ADMM/blob/main/ADMM_tf_vs_np.ipynb)). 

Although early stopping can resolve this issue, there might be a better set of hyper-parameters that lead to convergence. Hence, the aim is to tune the ADMM update hyper-parameters in such a way that the output converges to the optimal solution.

**Comment by utsav-akhaury** [comment]
- The **Gradient step size update (α)** is updated using the Spectral Norm (SV) of the operator, which is calculated using the [power iteration method](https://github.com/CosmoStat/score/blob/master/cadmos_lib.py#L176). 

- The update equation  -
  **α = 1 / ((SV + ρσ<sup>2</sup>) * (1 + 10<sup>-5</sup>))**

- The direct TensorFlow implementation (below) gives a similar output to the [power iteration method](https://github.com/CosmoStat/score/blob/master/cadmos_lib.py#L176) (the order of difference being ~10<sup>-4</sup>)
```    
H = tf.signal.fft2d(psf)
normH = tf.math.abs(tf.reverse(tf.reverse(H, axis=[0]), axis=[1]) * H + rho*sigma**2)
return tf.math.reduce_max(normH)
```


- The value of ρ (the augmented lagrangian parameter) does not change after a few iterations (because it reaches the upper cap defined in the [ADMM Deep Plug&Play code](https://github.com/CosmoStat/ShapeDeconv/blob/a18bf74fa7104f8621871c6ced61c503a4ec004a/python/DeepDeconv/admm/admm_FCS.py#L31)). Hence, SV & α do not change subsequently.

**Comment by utsav-akhaury** [comment]
Convergence achieved by updating the Gradient step size update (α) as discussed in the [comment](https://github.com/utsav-akhaury/ADMM/issues/3#issuecomment-779766371) mentioned above
Both NumPy and TensorFlow give similar results (order of difference ~10-4)
